code/modules/mob/living/living_say.dm

```diff
@@ -294,13 +294,23 @@ GLOBAL_LIST_EMPTY(channel_to_radio_key)
 	else //Turf, leave speech bubbles to the mob
 		speech_bubble("[bubble_icon][speech_bubble_test]", src, speech_bubble_recipients)
 
-	for(var/obj/O in listening_obj)
-		spawn(0) // KILL THIS
-			if(O) //It's possible that it could be deleted in the meantime.
-				O.hear_talk(src, message_pieces, verb)
+	hear_message_obj(listening_obj, src, message_pieces, verb)
 
 	return TRUE
 
+/proc/hear_message_obj(list/listening_obj, mob/M, list/message_pieces, verbage)
+	var/list/transmited_channels = list()
+	for(var/obj/O in listening_obj)
+		spawn(0) // KILL THIS
+			if(O) // It's possible that it could be deleted in the meantime.
+				if(isradio(O))
+					var/obj/item/radio/radio = O
+					if(radio.broadcasting && get_dist(radio, M) <= radio.canhear_range && !(radio.frequency in transmited_channels))
+						if(radio.talk_into(M, message_pieces, null, verbage))
+							transmited_channels += radio.frequency
+				else
+					O.hear_talk(M, message_pieces, verbage)
```

Comment:
```suggestion
	for(var/obj/O in listening_obj)
		spawn(0) // KILL THIS
		if(!istype(O)) // It's possible that it could be deleted in the meantime.
			return
		if(!isradio(O))
			O.hear_talk(M, message_pieces, verbage)
			return
		var/obj/item/radio/radio = O
		if(!radio.broadcasting || get_dist(radio, M) > radio.canhear_range || (radio.frequency in transmited_channels))
			return
		if(radio.talk_into(M, message_pieces, null, verbage))
			transmited_channels += radio.frequency
			
```

Tried to tidy this up, please test to make sure I've not made any mistakes

Reply:
The issue seems to be that the code `returns`, while it should `continue` instead, could you try that?